grafana/RabbitMQ Queues Overview - Seventh State RabbitMQ Support.json

```diff
@@ -372,14 +261,14 @@
           },
           "editorMode": "code",
           "exemplar": false,
-          "expr": " sum by (rabbitmq_cluster, vhost, queue) (rabbitmq_detailed_queue_messages{vhost=\"$rabbitmq_vhost\"} * on(instance, job) group_left(rabbitmq_cluster, rabbitmq_node) rabbitmq_identity_info{rabbitmq_cluster=\"$rabbitmq_cluster\", namespace=\"$namespace\"})\n",
+          "expr": " sum by (rabbitmq_cluster, vhost, queue) (rabbitmq_detailed_queue_messages{vhost=\"$rabbitmq_vhost\"} * on(instance, job) group_left(rabbitmq_cluster, rabbitmq_node) rabbitmq_identity_info{rabbitmq_cluster=\"$rabbitmq_cluster\", namespace=\"$namespace\"})",
```

Comment:
I see that that all of them use the same `rabbitmq_detailed_queue_messages` metric, is that required for some reason?

Reply:
The metric `rabbitmq_detailed_queue_messages` returns information about consumers on all queues, including those with no associated channels. With `rabbitmq_detailed_queue_messages * 0`, the resulting metric values are forced to 0. This allows me to extract only the queue names without any consumer data. Using the `or` operation, I combine the results of the "main" metric with the complementary metric that includes queues without channels. As a result, I can have a complete set of all queues, where queues without channels are represented with a 0 value.

Reply:
Thanks! It's a bit weird but I think it makes sense. 🙂